Update SL skill to include old spec too (#47)
dbt-labs/dbt-agent-skills: skills/building-dbt-semantic-layer/references/best-practices.md

## Core Principles

1. **Prefer normalization** - Let MetricFlow denormalize dynamically for end users rather than pre-building wide tables
2. **Compute in metrics, not rollups** - Define calculations in metrics instead of frozen aggregations
3. **Start simple** - Build simple metrics first before advancing to ratio and derived types

## Semantic Model Design

### Entities
- Each semantic model needs exactly **one primary entity**
- Use singular naming (`order` not `order_id`) with `expr` for the column reference
- Foreign entities enable joins between semantic models

### Dimensions
- Always include a **primary time dimension** when the model has metrics or measures
- Set granularity appropriately for time dimensions
- Use computed expressions for derived dimensions (e.g., categorizing by thresholds)

### Measures (Legacy Spec) / Simple Metrics (Latest Spec)

**Legacy spec** (dbt Core 1.6-1.11):
- Create measures for quantitative values you'll aggregate
- Use `expr: 1` with `agg: sum` for counting records
- Measures are the building blocks for all metric types
- Define components consistently: **entities -> dimensions -> measures**

**Latest spec** (dbt Core 1.12+ / Fusion):

> - Define simple metrics directly on the semantic model for quantitative aggregations

Author: ```suggestion
- Define simple metrics directly on the model for quantitative aggregations
```

| Anti-pattern | Better approach |
|--------------|-----------------|

> | Building semantic models on dimension-only tables without metrics | Only add primary entity for pure dimensional tables |

Author: ```suggestion
| Building full semantic models on dimension-only tables | Pure dimensional tables only need a primary entity defined |
```
